skbio/embedding/_embedding.py

```diff
@@ -98,3 +108,158 @@ def __repr__(self):
             f"embedding dimension: {dim}\n{indent}has gaps",
         )
         return rstr
+
+
+class SequenceVector(Embedding):
+    r"""Store a vector representation for a biological sequence."""
+    def __init__(self, vector, sequence, **kwargs):
+
+        if isinstance(sequence, Sequence):
+            sequence = str(sequence)
+        if isinstance(sequence, str):
+            sequence = sequence.encode("ascii")
+
+        if len(vector.shape) == 1:
+            vector = vector.reshape(1, -1)
+        if len(vector.shape) == 2:
+            assert vector.shape[0] == 1, (
+                "Only 1 vector per sequence is allowed."
+            )
+
+        seq = np.array([sequence], dtype='O')
+        super(SequenceVector, self).__init__(vector, seq, **kwargs)
+
+    def __str__(self):
+        return str(self._ids[0].decode('ascii'))
```

Comment:
The `decode` here, with the `encode on line 60, is essentially:

`b'asdasd'.decode('ascii').encode('ascii')`

...right? Is there room to reduce or remove some of the decode/encode operations?

Reply:
erm ... yes I think we can rid of the outer `str` call, since the input is already a `str`

Reply:
I wonder if we may pass sequences as byte arrays under the hood among `Sequence`, `Embedding` and `Vector`, since this is the native format for `Sequence`, and it is efficient. On the other hand, `str` is less native or efficient.

Reply:
We're currently passing around a numpy of str. This is what is done in the `DistanceMatrix` object, so I'm in favor of currently keeping this to avoid a major refactor here